lib/msf/core/post/windows/file_system.rb

```diff
@@ -292,17 +264,93 @@ def create_symlink(_root, link_name, target_name)
+        # command when *directory* is set to true.
+        #
+        # @param [String] link_name The path at which to create the symbolic link.
+        # @param [String] target_name The path that the new symbolic link targets.
+        # @param [Boolean] directory Whether or not the link target is a directory.
+        #
+        # @return [true, nil] Returns true on success or nil on failure.
+        def create_symlink(link_name, target_name, directory: true)
+          flags = directory ? 'SYMBOLIC_LINK_FLAG_DIRECTORY' : ''
+          result = session.railgun.kernel32.CreateSymbolicLinkW(link_name, target_name, flags)
+          unless result['GetLastError'] == SUCCESS
+            print_error("Error creating the symlink. Windows Error Code: #{result['GetLastError']} - #{result['ErrorMessage']}")
+            return nil
+          end
+
+          true
+        end
+
+        # Create a "Volume Mount Point" or a "Directory Junction". The difference between the two is that a Directory
+        # Junction targets a subdirectory of another volume where as a Volume Mount Point targets the root of a volume.
+        # This function is a suitable replacement for the `mklink /J` shell command.
+        #
+        # @see https://en.wikipedia.org/wiki/NTFS_reparse_point#Volume_mount_points
+        # @see https://nixhacker.com/understanding-and-exploiting-symbolic-link-in-windows/
+        #
+        # @param [String] path The path of where to place the mount point. This path must be an existing, empty directory.
+        # @param [String] target The target of what to mount at the specified path.
+        # @param [String] print_name The optional print name string.
+        #
+        # @return [Integer, nil] The handle to the reparse point which should be kept for use with {#delete_mount_point}
+        #   or nil on failure.
+        def create_mount_point(path, target, print_name = '')
+          return nil if target.empty? || path.empty?
+
+          fixed_target = target.start_with?('\\') ? target : "\\??\\#{target}"
+          reparse_data = build_reparse_data_buffer(fixed_target, print_name)
+
+          handle = open_reparse_point(path, true)
+          return nil unless handle
+
+          set_reparse_point(handle, reparse_data.to_binary_s)
+          handle
+        end
+
+        alias create_junction create_mount_point
+
+        private
+
+        def write_to_memory(process, str)
+          p_buffer = process.memory.allocate(str.size)
+          unless p_buffer
+            print_error("Error allocating memory for \"#{str}\": Windows Error Code: #{result['GetLastError']} - #{result['ErrorMessage']}")
+            return nil
+          end
+          unless process.memory.write(p_buffer, str) == str.size
+            print_error("Error writing \"#{str}\" to memory buffer: Windows Error Code: #{result['GetLastError']} - #{result['ErrorMessage']}")
+            return nil
+          end
```

Comment:
Isn't `result` undefined here? If it is defined can you point out the code where it is defined as right now it seems this code will error out on an undefined variable if `p_buffer` is invalid.